app/apikey.go

```diff
@@ -0,0 +1,231 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/temporalio/tcld/protogen/api/auth/v1"
+	"github.com/temporalio/tcld/protogen/api/authservice/v1"
+	"github.com/urfave/cli/v2"
+	"google.golang.org/grpc"
+)
+
+type APIKeyClient struct {
+	authClient authservice.AuthServiceClient
+	ctx        context.Context
+}
+
+func NewAPIKeyClient(ctx context.Context, conn *grpc.ClientConn) *APIKeyClient {
+	return &APIKeyClient{
+		authClient: authservice.NewAuthServiceClient(conn),
+		ctx:        ctx,
+	}
+}
+
+type GetAPIKeyClientFn func(ctx *cli.Context) (*APIKeyClient, error)
+
+func GetAPIKeyClient(ctx *cli.Context) (*APIKeyClient, error) {
+	ct, conn, err := GetServerConnection(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return NewAPIKeyClient(ct, conn), nil
+}
+
+func (s *APIKeyClient) createAPIKey(
+	displayName string,
+	description string,
+	expiry time.Time,
+	requestID string,
+) error {
+	expiryts, err := types.TimestampProto(expiry)
+	if err != nil {
+		return fmt.Errorf("failed to convert timestamp to proto: %s", err)
+	}
+	resp, err := s.authClient.CreateAPIKey(s.ctx, &authservice.CreateAPIKeyRequest{
+		Spec: &auth.APIKeySpec{
+			DisplayName: displayName,
```

Comment:
I think in the UX design we said the display name is required, could be good to do a quick length check here

Reply:
The displayname is set as `required`. urfave/cli package will make sure its never empty. Not point of adding a redundant check.